packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -1221,6 +1249,27 @@ export class UserPool extends UserPoolBase {
         throw new Error(`Unsupported AccountRecovery type - ${accountRecovery}`);
     }
   }
+
+  private configureUserAttributeChanges(props: UserPoolProps) {
+    let userAttributeUpdateSettings;
+    if (props.keepOriginal) {
+      let attributesRequireVerificationBeforeUpdate: string[] = [];
+
+      if (props.keepOriginal.email) {
+        attributesRequireVerificationBeforeUpdate.push('email');
+      }
+
+      if (props.keepOriginal.phone) {
+        attributesRequireVerificationBeforeUpdate.push('phone');
+      }
+
+      userAttributeUpdateSettings = {
+        attributesRequireVerificationBeforeUpdate,
+      };
+    }
+
+    return userAttributeUpdateSettings;
+  }
```

Comment:
(Optional to address)

This is just personal style, but I prefer negative guards to avoid nesting ifs and improve readability:

```suggestion
    if (!props.keepOriginal) {
      return undefined;
    }
    
    const attributesRequireVerificationBeforeUpdate: string[] = [];
    
    if (props.keepOriginal.email) {
      attributesRequireVerificationBeforeUpdate.push('email');
    }
    if (props.keepOriginal.phone) {
      attributesRequireVerificationBeforeUpdate.push('phone');
    }
    
    return {
      attributesRequireVerificationBeforeUpdate,
    };
  }
```

Reply:
I am a fan of this way of coding too! I will change it